src/hull.jl

```diff
@@ -1,163 +1,246 @@
-CI = MOI.ConstraintIndex
-SAF = MOI.ScalarAffineFunction
-
-"""
-    hull_reformulation!(constr::ConstraintRef{<:AbstractModel, CI{SAF{T},V}}, bin_var, args...) where {T,V}
-
-Apply the hull reformulation to a linear constraint.
-
-    hull_reformulation!(constr::ConstraintRef, bin_var, eps, i, j, k)
-
-Apply the hull reformulation to a nonlinear constraint (includes quadratic) at index k of constraint j in disjunct i.
-
-    hull_reformulation!(constr::AbstractArray{<:ConstraintRef}, bin_var, eps, i, j, k)
+################################################################################
+#                              VARIABLE DISAGGREGATION
+################################################################################
+function _query_variable_bounds(model::JuMP.Model, method::Hull)
+    for var in JuMP.all_variables(model)
+        method.variable_bounds[var] = _update_variable_bounds(var)
+    end
+end
 
-Call the hull reformulation on a constraint at index k of constraint j in disjunct i.
-"""
-function hull_reformulation!(constr::ConstraintRef{<:AbstractModel, CI{SAF{T},V}}, bin_var, args...) where {T,V}
-    #check constraint type
-    i = args[2] #get disjunct index
-    bin_var_ref = constr.model[bin_var][i]
-    #replace each variable with its disaggregated version
-    for var_ref in constr.model[:gdp_variable_refs]
-        #get disaggregated variable reference
-        var_name_i = name_disaggregated_variable(var_ref, bin_var, i)
-        var_i_ref = variable_by_name(constr.model, var_name_i)
-        #check var_ref is present in the constraint
-        coeff = normalized_coefficient(constr, var_ref)
-        iszero(coeff) && continue #if not present, skip
-        #swap variable for disaggregated variable
-        set_normalized_coefficient(constr, var_ref, 0) #remove original variable
-        set_normalized_coefficient(constr, var_i_ref, coeff) #add disaggregated variable
+function _update_variable_bounds(vref::JuMP.VariableRef)
+    if JuMP.is_binary(vref) #not used
+        lb, ub = 0, 1
+    elseif !JuMP.has_lower_bound(vref) || !JuMP.has_upper_bound(vref)
+        error("Variable $vref must have both lower and upper bounds defined when using the Hull reformulation.")
+    else
+        lb = min(0, JuMP.lower_bound(vref))
+        ub = max(0, JuMP.upper_bound(vref))
+    end
+    return lb, ub
+end
+function _disaggregate_variables(model::JuMP.Model, lvref::LogicalVariableRef, vrefs::Set{JuMP.VariableRef}, method::_Hull)
+    #create disaggregated variables for that disjunct
+    for vref in vrefs
+        JuMP.is_binary(vref) && continue #skip binary variables
```

Comment:
What about integer variables?

Reply:
GDP currently assumes all variables are continuous. Integer variables should never show up in GDP, although there is some work by John Hooker where they use mixed-integer constraints inside of the disjuncts. We only skip the binary variables to accommodate for when we have nested disjunctions since the inner reformulated constraints will have binary variables.

Reply:
We should probably have some more careful checks then to assert these conditions

Reply:
Like we should check the variables of disjunct constraints before we add them to ensure they do not contain integer/binary variables

Reply:
Ok. I'll add these checks in the `build_constraint` for the disjunct constraints

Reply:
done